### PR TITLE
:shipit: Assigment 2 - Code Refactoring

### PR DESCRIPTION
**Pull request by ray2g**
📑 **Description**
---

The function `clean_data` was refactored in three in separated function: `load_data`, `clean_data`, and `save_data`, where each function as a unique task.

- `load_data` function should return the data. 
- `clean_data` function takes the data as an argument and return the cleaned data.  
- `save_data` function takes the data as an argument and save it to a file.

### CODE REVIEW

**Review by jpleitao — COMMENTED** [review]
Minor suggestions based on python's style guide. Main point related to the absence of type conversion for columns _year_ and _value_ during the DataFrame cleaning (as we are not making any modification to these columns, it likely has no impact on the result of the tests). Otherwise code looks good and passes on the tests!

File: assignments/life_expectancy/cleaning.py [hunk 1]
```diff
@@ -74,18 +90,28 @@ def clean_data(input_path: Union[str, Path], output_path: Union[str, Path], regi
     df_clean = df_unpivot.dropna()
     
     # filter dataframe by desire region
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/cleaning.py:92]: It is performed in the code lines from 83 to 87:

```
# convert value column into float
df_unpivot['value'] = df_unpivot['value'].apply(pd.to_numeric, errors="raise")

# convert "year" column to integer
df_unpivot['year'] = df_unpivot['year'].astype(int)
```

File: assignments/life_expectancy/cleaning.py [hunk 2]
```diff
@@ -2,40 +2,57 @@
...
 # import libraries
 import io
+from io import TextIOWrapper
 import csv
 from pathlib import Path
 from typing import Union
 import argparse
 import pandas as pd
+from pandas import DataFrame
 import numpy as np
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/cleaning.py:9]: fixed

File: assignments/life_expectancy/cleaning.py [hunk 3]
```diff
@@ -74,18 +90,28 @@ def clean_data(input_path: Union[str, Path], output_path: Union[str, Path], regi
...
 
     print(f"Process Finished! The file was saved as csv in:\n{output_path}\n")
 
 
 def main():
     """
     Main function which config the arparse and
-    calls the function to clean the tsv file.
+    calls the functions to load, clean and save the tsv file.
     """
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/cleaning.py:132]: fixed

File: assignments/life_expectancy/cleaning.py [hunk 4]
```diff
@@ -94,8 +120,14 @@ def main():
     parser.add_argument("-r", default="PT", help = "Desire region to filter the csv.")
     args = parser.parse_args()
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/cleaning.py:139]: fixed

File: assignments/life_expectancy/cleaning.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpleitao) [reply assignments/life_expectancy/cleaning.py:97]: My bad, I missed it (maybe was not a change in this MR and I did not noticed it in the entire file). Point is closed

**Pull request merged**